packages/block-editor/src/components/block-edit/index.js

```diff
@@ -34,12 +37,42 @@ export default function BlockEdit( props ) {
 		'__experimentalLayout',
 		false
 	);
+	const shouldDisplayControls = useSelect(
+		( select ) => {
+			if ( isSelected ) {
+				return true;
+			}
+
+			const {
+				getBlockName,
+				isFirstMultiSelectedBlock,
+				getMultiSelectedBlockClientIds,
+				hasSelectedInnerBlock,
+			} = select( blockEditorStore );
+
+			if ( isFirstMultiSelectedBlock( clientId ) ) {
+				return getMultiSelectedBlockClientIds().every(
+					( id ) => getBlockName( id ) === name
+				);
+			}
+
+			return (
+				hasBlockSupport(
+					getBlockName( clientId ),
+					'__experimentalExposeControlsToChildren',
+					false
+				) && hasSelectedInnerBlock( clientId )
+			);
+		},
+		[ clientId, isSelected, name ]
+	);
 	const context = {
```

Comment:
Is this context a public API, if it is, I think we should "lock" the new key somehow to make it private.

Reply:
How do you "lock" a key?

I'm thinking we should maybe create a separate boolean context for this that is private to blockEditor which can be used in the Fills and pass it down as a prop to BlockEdit for the HoCs so they don't need to call a hook.

Reply:
yeah either a separate context or store a locked function that returns the boolean.

Reply:
I probably prefer the former to avoid the proliferation of contexts

Reply:
Why should we lock it though? Don't we want plugin authors to use these in their filters?